src/nvim/path.c

```diff
@@ -373,10 +373,6 @@ void add_pathsep(char *p)
 char *FullName_save(char *fname, bool force)
   FUNC_ATTR_NONNULL_RET FUNC_ATTR_MALLOC FUNC_ATTR_NONNULL_ALL
 {
-  if (fname == NULL) {
-    return NULL;
-  }
```

Comment:
Sorry if I advised this, but unless we can prove that `FullName_save()` never gets called with a `NULL` pointer, removing this isn't safe and the `FUNC_ATTR_NONNULL_RET` must be removed to prevent invalid optimizations.